### PR TITLE
First Version

### PR DESCRIPTION
**Pull request by jaspal007**
## About
- **Info about the issue:**  This PR is to have a basic login verification screen on for an application. 
- **Name:** Jaspal Singh
- **GitHub ID:**  jaspal007

## Approach
-  For now the package I used was the default flutter package  `flutter/material.dart`
-  Used Column widget for Listing all the other widgets.
-  Used `enums` for the `Elevated Button` that shows whether email or SMS mode is selected.
-  Used `resizeToAvoidBottomInset` to check the pixel overflow when keyboard is enabled.

## Screenshot

Page1

Page2

### CODE REVIEW

**Review by RitwikSingh28 — CHANGES_REQUESTED** [review]
@jaspal007 
Delete the .git folder (hidden) in your machine's remote repo, and then try to push the changes to this PR. 
You need to make sure that your project folder is not a git repository in itself. As the outer folder is already a git repo. Do look after it and make changes accordingly.

**Comment by jaspal007** [comment]
> @jaspal007 Delete the .git folder (hidden) in your machine's remote repo, and then try to push the changes to this PR. You need to make sure that your project folder is not a git repository in itself. As the outer folder is already a git repo. Do look after it and make changes accordingly.

I have reviewed my file on my local machine and performed the task as stated.

**Comment by jaspal007** [comment]
Reopening my closed PR

File: prac2-pages/Jaspal/login.dart [hunk 1]
```diff
@@ -0,0 +1,146 @@
+import 'package:flutter/material.dart';
+
+class Login extends StatelessWidget {
+  bool firstvalue = false;
+  final Function check;
+  final Function selectHandler;
+
+  Login(
+      {super.key,
+      required this.selectHandler,
```
Comment (RitwikSingh28) [inline prac2-pages/Jaspal/login.dart]: This method of delegating calls to the parent is actually quite a nice touch! It follows good software design principles, (separation of concerns)... :sunglasses:

File: prac2-pages/Jaspal/login.dart [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                        RoundedRectangleBorder(
+                          borderRadius: BorderRadius.circular(10),
+                        ),
+                      ),
+                      backgroundColor:
+                          MaterialStateProperty.all<Color>(Colors.black),
+                      foregroundColor:
+                          MaterialStateProperty.all<Color>(Colors.white),
+                    ),
+                    onPressed: () => 3,
```
Comment (RitwikSingh28) [inline prac2-pages/Jaspal/login.dart]: You can try to perform the following tasks here:

- Perform form validation (basic check if they are empty or email contains @)
- If passed, redirect the user to the second page.
- Try to learn about Navigation in Flutter, and use that to route the user to a different page

File: prac2-pages/Jaspal/main.dart [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    });
+  }
+
+  @override
+  build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        resizeToAvoidBottomInset: false,
+        body: Center(
+          child: (_tap == false)
```
Comment (RitwikSingh28) [inline prac2-pages/Jaspal/main.dart]: This method is creative, but will result in a lot of overhead in performance. For routing between the pages, it is advisable to use **Navigation**. 
Kindly remove this and try to implement page routing here.

File: prac2-pages/Jaspal/main.dart [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+              ? Column(
```
Comment (RitwikSingh28) [inline prac2-pages/Jaspal/main.dart]: **main.dart file should only contain the logic for deciding the initial screens**. The pages to be displayed should be decided here, and not their implementations. :information_source: 

Kindly refactor this into a widget, and put it in a separate file. And then provide its instance here. :thinking: 

Makes the code readable and easily maintainable. :thought_balloon:

**Review by RitwikSingh28 — CHANGES_REQUESTED** [review]
Great work so far @jaspal007 !! 

Please implement the fixes to the aforementioned issues, and then push the fixes with subsequent commits.

**Note**: Try to fix one issue in a single commit. Results in better readability and version control.

File: prac2-pages/Jaspal/login.dart (hunk 2 above)
Comment (jaspal007) [reply prac2-pages/Jaspal/login.dart]: Recreated the login widget with form validation

File: prac2-pages/Jaspal/main.dart (hunk 3 above)
Comment (jaspal007) [reply prac2-pages/Jaspal/main.dart]: Rebuilt the app with navigator that uses context stack

File: prac2-pages/Jaspal/main.dart (hunk 4 above)
Comment (jaspal007) [reply prac2-pages/Jaspal/main.dart]: Refactored the main.dart to only contain the reference to the home screen and the MaterialApp

File: prac2-pages/Jaspal/login.dart (hunk 2 above)
Comment (RitwikSingh28) [reply prac2-pages/Jaspal/login.dart]: Perfectly done! We will be using regex a lot in our upcoming projects!

File: prac2-pages/Jaspal/main.dart (hunk 4 above)
Comment (RitwikSingh28) [reply prac2-pages/Jaspal/main.dart]: Resolved! :+1:

**Review by RitwikSingh28 — CHANGES_REQUESTED** [review]
Everything is perfect @jaspal007 ! 

One other improvement that I would like you to now make is to make the UI adaptive on different mobile displays. Since the values in px are specified as per your test screen, there is a chance that the UI might overflow in other screens.

*Hint* :thought_balloon: 
Look up ScreenUtil package in Flutter

File: prac2-pages/Jaspal/main.dart (hunk 3 above)
Comment (RitwikSingh28) [reply prac2-pages/Jaspal/main.dart]: Resolved

**Review by RitwikSingh28 — APPROVED** [review]
Fantastic @jaspal007 :100: 

Everything is working perfectly, and all the improvements requested have been made. 
I hope with this task, you got a feel of how it looks like to contribute to an open-source project. :smile: 

Merging your PR :+1:

**Pull request merged**